### PR TITLE
Add R/W implementation of capture file

### PR DESCRIPTION
**Pull request by dimitry-**
Add CaptureFile implementation to work with sectioned file.
Also update file format document accordingly.

Test: run CaptureFileTests

### CODE REVIEW

File: src/CaptureFile/CaptureFile.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  }
+
+  // Take a copy of section list
+  auto section_list = section_list_;
+
+  // The current file format always places the section list is at the end of the file
+  // therefore the current section_offset is going to be the offset for the new section.
+  section_list.push_back(CaptureFileSection{/*.type = */ section_type,
+                                            /*.offset = */ header_.section_list_offset,
+                                            /*.size = */ section_size});
```
Comment (dimitry-) [inline src/CaptureFile/CaptureFile.cpp:134]: well, this is quite a bug... not sure why tests did not catch it - converted to draft. will need another review probably tomorrow.

Comment (dimitry-) [reply src/CaptureFile/CaptureFile.cpp:147]: fixed(implementation and the test)

**Comment by dimitry-** [comment]
ptal

File: src/CaptureFile/include/CaptureFile/CaptureFileSection.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// found in the LICENSE file.
+
+#ifndef CAPTURE_FILE_CAPTURE_FILE_SECTION_H_
+#define CAPTURE_FILE_CAPTURE_FILE_SECTION_H_
+
+#include <cstdint>
+
+namespace orbit_capture_file {
+
+struct CaptureFileSection {
```
Comment (florian-kuebler) [inline src/CaptureFile/include/CaptureFile/CaptureFileSection.h:12]: Not sure if you want to add a super brief comment here. Feel free to ignore though.

File: src/CaptureFile/include/CaptureFile/CaptureFile.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#include <filesystem>
+#include <memory>
+#include <vector>
+
+#include "CaptureFile/CaptureFileSection.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_capture_file {
+
+class CaptureFile {
```
Comment (florian-kuebler) [inline src/CaptureFile/include/CaptureFile/CaptureFile.h:17]: Could you add a brief class-level comment here?

File: src/CaptureFile/CaptureFile.cpp [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        "Unexpected EOF while reading section list: section list size=%d, bytes available=%d",
+        number_of_sections * sizeof(CaptureFileSection), bytes_read)};
+  }
+
+  section_list_ = std::move(section_list);
+
+  return outcome::success();
+}
+
+const std::vector<CaptureFileSection>& CaptureFileImpl::GetSectionList() const {
```
Comment (florian-kuebler) [inline src/CaptureFile/CaptureFile.cpp:95]: Personal preference (feel free to ignore): I would implement the getter already at the definition above.

**Comment by dpallotti** [comment]
> ptal

I won't be able to take a look before Monday (in case you want to add another reviewer).

File: src/CaptureFile/CaptureFileTest.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    std::string content;
+    content.resize(something.size());
+    auto read_result =
+        capture_file->ReadFromSection(0, kOffsetInSection, content.data(), something.size());
+    ASSERT_FALSE(read_result.has_error()) << read_result.error().message();
+    EXPECT_EQ(content, something);
+  }
+}
+
+TEST(CaptureFile, OpenInvalidCaptureFile) {
```
Comment (florian-kuebler) [inline src/CaptureFile/CaptureFileTest.cpp:112]: There are couple of reasons for a file being invalid, right? Should we have a test for each reason?

File: src/CaptureFile/include/CaptureFile/CaptureFile.h (hunk 3 above)
Comment (dimitry-) [reply src/CaptureFile/include/CaptureFile/CaptureFile.h:21]: done.

**Review by florian-kuebler — APPROVED** [review]
LGTM, thanks!

**Pull request merged**